src/net.rs

```diff
@@ -17,19 +17,123 @@
 
 // TODO: Remove this line as soon as this module is actually used
 #![allow(dead_code, unused_variables)]
+use std::net::TcpStream;
+use std::io::{Write,Read, Error, ErrorKind};
+use byteorder::{BigEndian, WriteBytesExt, ReadBytesExt}; // for write_u16()
+use std::io;
+use bincode::rustc_serialize::{decode_from, encode_into};
+use bincode::SizeLimit;
+use std::io::Cursor;
+
+#[derive(RustcEncodable, RustcDecodable)]
+#[repr(u8)]
+pub enum CNV{
+	GREETING = 0,
+	LOGIN,
+	ERROR,
+}
+#[derive(RustcEncodable, RustcDecodable)]
+#[repr(u8)]
+pub enum ErrorCode{
+	UNSPECIFIC_ERR = 0,
+	IO_ERR,
+	UNEXPEC_PKG_ERR,
+}
 
 const PROTOCOL_VERSION : u8 = 1;
 
 /// This is the first packet being sent by the server after the TCP connection
 /// is established.
+#[derive(RustcEncodable, RustcDecodable)]
 pub struct Greeting {
-    protocol_version: u8,
+    protocol_version: u8,	// 1 byte
+    message: String,		// n bytes
+}
+
+impl Greeting{
+	pub fn make_greeting(version: u8, msg: String)-> Greeting{
+		Greeting{protocol_version: version, message: msg}
+	}
+}
+
+/// writes a welcome-message to the given server-client-stream
+pub fn do_handshake<W:Write + Read>(stream: &mut W) -> Result<(String, String), io::Error>{
+	let greet = Greeting::make_greeting(PROTOCOL_VERSION, "Welcome".to_string());
```

Comment:
Just as a note (you don't need to change it): You could use `into()` instead of `to_string()` here.